docs/pages/blog/masonry-post.js

```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import TopLayoutBlog from 'docs/src/modules/components/TopLayoutBlog';
+import { docs } from './masonry-post.md?@mui/markdown';
```

Comment:
I'm not sure about this URL slug, blog is implicitly a post.

```suggestion
import { docs } from './building-a-masonry.md?@mui/markdown';
```

Maybe this URL, using https://react-spectrum.adobe.com/blog/building-a-button-part-1.html as inspiration?